opm/input/eclipse/Schedule/Schedule.hpp

```diff
@@ -335,6 +336,14 @@ namespace Opm
             this->template pack_unpack_map<int, VFPInjTable>(serializer);
             this->template pack_unpack_map<std::string, Group>(serializer);
             this->template pack_unpack_map<std::string, Well>(serializer);
+
+            if (!serializer.isSerializing()) {
+                for (auto& snapshot : snapshots) {
+                    for (auto& well : snapshot.wells) {
+                        well.second->updateUnitSystem(&m_static.m_unit_system);
+                    }
+                }
```

Comment:
You probably know this already, but in the interest of completeness I'll just remark that this is safe in this particular context since every call to `Well::updateUnitSystem()` passes the same pointer value.  In general, however, this kind of loop structure is potentially harmful.  Since the snapshots store `shared_ptr<>` objects, we risk making inconsistent changes to both earlier and later `Well` objects if we're not careful and that's why most `update()` functions return a `bool` to let the caller know whether or not any actual changes were made.

Reply:
Yeah, I am aware. As it is the same pointer being put into every well this is non-problematic and I need to do this after deserialization since the pointer obviously is process specific.